### ISSUE TITLE
segmentation fault using predict() !

### ISSUE THREAD

**Issue by courtiol** [glmmTMB/glmmTMB#332]
Dear all,

I spotted a critical bug.

The following code leads to a segmentation fault crashing R on one of my (otherwise healthy) system but not on another one (the crash could be random but I replicated the same outcome 3 times on each system). The session infos of the two systems are attached below.

Note that on the system for which it runs, the outcome of predict() varies between successive call, which also points toward a memory management issue.

Best

Alex

PS: I used data from https://en.wikipedia.org/wiki/Beta-binomial_distribution

```{r}
d <- data.frame(Males = 0:12,
                           Females = 12:0,
                           Families = c(3, 24, 104, 286, 670, 1033, 1343, 1112, 829, 478, 181, 45, 7))
dd <- d[rep(1:nrow(d), d$Families), ]
dd$Families <- NULL
dd$ID <- factor(1:nrow(dd))

library(glmmTMB)
mod <- glmmTMB(cbind(Males, Females) ~ 1 + (1|ID), data = dd, family = betabinomial())
predict(mod)
```

Using R inside a terminal gives some information about the issue:

```{r}
 *** caught segfault ***
address 0x1109b5000, cause 'memory not mapped'

Traceback:
 1: retape()
 2: MakeADFun(data.tmb, parameters, map = mapArg, random = randomArg,     profile = NULL, silent = TRUE, DLL = "glmmTMB")
 3: eval(substitute(expr), data, enclos = parent.frame())
 4: eval(substitute(expr), data, enclos = parent.frame())
 5: with.default(TMBStruc, MakeADFun(data.tmb, parameters, map = mapArg,     random = randomArg, profile = NULL, silent = TRUE, DLL = "glmmTMB"))
 6: with(TMBStruc, MakeADFun(data.tmb, parameters, map = mapArg,     random = randomArg, profile = NULL, silent = TRUE, DLL = "glmmTMB"))
 7: predict.glmmTMB(mod)
 8: predict(mod)
```

Session info for the system that crashes:

```{r}
> sessionInfo()
R version 3.5.0 (2018-04-23)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS High Sierra 10.13.4

Matrix products: default
BLAS: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] glmmTMB_0.2.0

loaded via a namespace (and not attached):
 [1] minqa_1.2.4     MASS_7.3-50     compiler_3.5.0  Matrix_1.2-14   tools_3.5.0     TMB_1.7.13      yaml_2.1.19    
 [8] Rcpp_0.12.16    splines_3.5.0   nlme_3.1-137    grid_3.5.0      nloptr_1.0.4    lme4_1.1-17     lattice_0.20-35
```

Session info for the system that does not crash but gives inconsistent results:
```{r}
> sessionInfo()
R version 3.4.4 (2018-03-15)
Platform: x86_64-redhat-linux-gnu (64-bit)
Running under: Scientific Linux release 6.9 (Carbon)

Matrix products: default
BLAS: /usr/lib64/R/lib/libRblas.so
LAPACK: /usr/lib64/R/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C
 [3] LC_TIME=en_GB.UTF-8        LC_COLLATE=en_GB.UTF-8
 [5] LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_GB.UTF-8
 [7] LC_PAPER=en_GB.UTF-8       LC_NAME=C
 [9] LC_ADDRESS=C               LC_TELEPHONE=C
[11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base

other attached packages:
[1] glmmTMB_0.2.0

loaded via a namespace (and not attached):
 [1] minqa_1.2.4     MASS_7.3-49     compiler_3.4.4  Matrix_1.2-12
 [5] tools_3.4.4     TMB_1.7.13      Rcpp_0.12.16    splines_3.4.4
 [9] nlme_3.1-131.1  grid_3.4.4      nloptr_1.0.4    lme4_1.1-16
[13] lattice_0.20-35 tcltk_3.4.4
```

**Comment by kaskr** [comment]
Replicate if #289 ?

@courtiol Does upgrading to the github version solve your issue ?

@mebrooks Perhaps time to do a new CRAN submission soon ?

**Comment by courtiol** [comment]
You are right @kaskr !  I had searched through old issues using "segmentation fault" as key words but had found nothing. Many thanks!

**Comment by mebrooks** [comment]
@kaskr, I agree that it's time for a CRAN submission. I wonder if we can agree on ways to fix #294 and #295 before the next release. I will read the discussions again and maybe schedule a skype meeting.
